src/widget/wlibrarytableview.cpp

```diff
@@ -92,36 +92,59 @@ void WLibraryTableView::moveSelection(int delta) {
     }
 
     while(delta != 0) {
-        // TODO(rryan) what happens if there is nothing selected?
-        QModelIndex current = currentIndex();
-        if(delta > 0) {
-            // i is positive, so we want to move the highlight down
-            int row = current.row();
-            if (row + 1 < pModel->rowCount()) {
-                selectRow(row + 1);
+        QItemSelectionModel* currentSelection = selectionModel();
+        if (currentSelection->selectedRows().length() > 0) {
+            if(delta > 0) {
+                // i is positive, so we want to move the highlight down
+                int row = currentSelection->selectedRows().last().row();
+                if (row + 1 < pModel->rowCount()) {
+                    selectRow(row + 1);
+                } else {
+                    selectRow(0);
+                }
+
+                delta--;
+            } else {
+                int row = currentSelection->selectedRows().first().row();
+                if (row - 1 >= 0) {
+                    selectRow(row - 1);
+                } else {
+                    selectRow(pModel->rowCount() - 1);
+                }
+
+                delta++;
             }
-
-            delta--;
         } else {
-            // i is negative, so we want to move the highlight up
-            int row = current.row();
-            if (row - 1 >= 0) {
-                selectRow(row - 1);
+            // no selection, so select the first or last element depending on delta
+            if(delta > 0) {
+                selectRow(0);
+                delta--;
+            } else {
+                selectRow(pModel->rowCount() - 1);
+                delta++;
             }
-
-            delta++;
         }
     }
 }
 
-void WLibraryTableView::saveVScrollBarPos(TrackModel* key){
-    m_vScrollBarPosValues[key] = verticalScrollBar()->value();
+void WLibraryTableView::saveVScrollBarPos(TrackModel* model){
+    if (model == nullptr) {
+        return;
+    }
+    qDebug() << "save: m_vScrollBarPosValues:" << model->key() << verticalScrollBar()->value() << " | " << m_vScrollBarPosValues;
+    m_vScrollBarPosValues[model->key()] = verticalScrollBar()->value();
 }
 
-void WLibraryTableView::restoreVScrollBarPos(TrackModel* key){
+void WLibraryTableView::restoreVScrollBarPos(TrackModel* model){
     updateGeometries();
+    if (model == nullptr) {
+        return;
+    }
+    QString key = model->key();
+    qDebug() << "m_vScrollBarPosValues:" << m_vScrollBarPosValues;
 
     if (m_vScrollBarPosValues.contains(key)){
+        qDebug() << "restore key found:" << key << " pos: " << m_vScrollBarPosValues[key];
         verticalScrollBar()->setValue(m_vScrollBarPosValues[key]);
     }else{
```

Comment:
```suggestion
    } else {
```
wouldn't pre-commit clang fix that?